src/pages/index.js

```diff
@@ -1,95 +1,245 @@
 
+function handleDeleteClick(card) {
+  const id = card.getID();
+  deleteCardModal.open(() => {
+    deleteCardModal.setLoading();
+    api
+      .removeCard(id)
+      .then(() => {
+        card.handleDeleteCard();
+        deleteCardModal.close();
+      })
+      .catch((error) => {
+        console.error("Error removing card:", error);
+      })
+      .finally(() => {
+        deleteCardModal.resetButtonText();
+      });
+  });
+}
+
+function handleAddLike(card) {
+  const id = card.getID();
+  api
+    .addLike(id)
+    .then((updatedCardData) => {
+      card.setLiked(true);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleRemoveLike(card) {
+  const id = card.getID();
+  api
+    .removeLike(id)
+    .then((updatedCardData) => {
+      card.setLiked(false);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
 function handleAddCardFormSubmit(inputValues) {
-  console.log(this._modalForm);
   const name = inputValues.title;
   const link = inputValues.url;
-  const newCardElement = createCard({ name, link });
-  cardsList.prependItem(newCardElement);
-  addCardModal.close();
-  addCardFormValidator.resetValidation();
-  addCardModal.reset();
+  addCardModal.setLoading();
+  api
+    .addCard({ name, link })
+    .then((newCardData) => {
+      const newCardElement = createCard(newCardData);
+      cardsList.prependItem(newCardElement);
+      addCardModal.close();
+      addCardFormValidator.resetValidation();
+      addCardModal.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardModal.resetButton();
+    });
+}
+
+function handleUpdateAvatarFormSubmit(inputValues) {
+  const link = inputValues.url;
+  updateAvatarModal.setLoading();
+  api
+    .changeProfilePicture(link)
+    .then((data) => {
+      userInfo.setAvatar(data.avatar);
+      updateAvatarModal.reset();
+      updateAvatarModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      updateAvatarModal.resetButton();
+    });
+}
+
+function handleUpdateAvatar() {
+  const link = document.getElementById("avatar-image-url").value;
+  updateAvatarModal.open(() => {
+    api
+      .changeProfilePicture(link)
+      .then(() => {
+        deleteCardModal.close();
+      })
+      .catch((error) => {
+        console.error("Error removing card:", error);
+      });
+  });
 }
 
 function handleProfileEditSubmit(inputValues) {
   userInfo.setUserInfo(inputValues);
+  profileEditPopup.setLoading();
+  api
+    .updateUserInfo(inputValues)
+    .then((userData) => {
+      userInfo.setUserInfo({
+        title: userData.name,
+        description: userData.about,
+      });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileEditPopup.resetButton();
+    });
   profileEditPopup.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well